src/Mvc/Mvc.ViewFeatures/src/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -206,8 +205,9 @@ internal static void AddViewServices(IServiceCollection services)
             services.TryAddSingleton<SaveTempDataFilter>();
 
             //
-            // Component prerendering
+            // Component rendering
             //
+            services.TryAddScoped<IComponentRenderer, ComponentRenderer>();
```

Comment:
Can we get rid of the interface an use an internal abstract class with a public virtual method instead?

Reply:
Is there a particular benefit to changing this? Given it's internal, we should be free to modify the contract in the future if we need to.